_rules/table-header-cells-refer-to-data-cells-d0f69e.md

```diff
@@ -0,0 +1,234 @@
+
+## Expectation
+
+Each target element is either the row header or the column header for data cells within the same `table`.
+
+## Assumptions
+
+- Tables are [well-formed, according to the HTML5.1 specification](https://www.w3.org/TR/html51/tabular-data.html#forming-a-table).
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 1.3.1: Information and relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
+- [H43: Using id and headers attributes to associate data cells with header cells in data tables](https://www.w3.org/WAI/WCAG21/Techniques/html/H43)
+- [Forming relationships between data cells and header cells](https://www.w3.org/TR/html/tabular-data.html#forming-relationships-between-data-cells-and-header-cells)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+`th` has corresponding `cells` within the same `table`.
+
+```html
+<table>
+	<tr>
+		<th>Time</th>
+	</tr>
+	<tr>
+		<td>05:41</td>
+	</tr>
+</table>
+```
+
+#### Passed Example 2
+
+`th` has corresponding `cells` within the same `table`. Here the cells span multiple columns.
+
+```html
+<table>
+  <thead>	
+    <tr>
+      <th>Projects</th>
+      <th>Exams</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td colspan="2">15%</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+#### Passed Example 3
+
+The `columnheader` has corresponding `cells` within the same `table`, by usage of `headers` attribute.
+
+```html
+<table>
+  <tr> 
+    <td headers="header1">
+      Projects
+    </td>
+    <th id="header1">
+      Projects
+    </th> 
+  </tr>
+</table>
+```
+
+#### Passed Example 4
+
+The `columnheader` has corresponding `cells` within the same `table`, by usage of `aria-labelledby` attribute.
+
+```html
+<table>
+	<tr>
+    <th id="hd2">Time</th>
+    <td aria-labelledby="hd2">05:44</td>
+	</tr>
+</table>
+```
+
+#### Passed example 5
+
+Each `th` element has corresponding `cells` within the same `table`, by usage of `scope` attribute.
+
+```html
+<table>
+  <caption>Hogwarts Houses stars</caption>
+  <tr>
+      <th scope="col">Name</th>
+      <th scope="col">House</th>
+  </tr>
+  <tr>
+      <th scope="row">Harry Potter</th>
+      <td>Gryffindor</td>
+  </tr>
+  <tr>
+      <th scope="row">Cedric Diggory</th>
+      <td>Hufflepuff</td>
+  </tr>
+  <tr>
+      <th scope="row">Cho Chang</th>
+      <td>Ravenclaw</td>
+  </tr>
+    <tr>
+      <th scope="row">Severus Snape</th>
+      <td>Slytherin</td>
+  </tr>
+</table>
+```
+
+### Failed
+
+#### Failed Example 1
+
+`th` elements within the `table` do not have corresponding `cells`.
+
+```html
+<table>
+	<tr>
+		<th>Header 1</th>
+	</tr>
+	<tr>
+		<th>Header 2</th>
+	</tr>
+</table>
+```
+
+#### Failed Example 2
+
+`th` does not have `cell`.
```

Comment:
`cell`.

Plus, there is no `th` element in that table, only a `td` element with a role of `columnheader`. Per current applicability, this is inapplicable.
* Do you mean that applicability should be "any element with a role of `columnheader`"?
* You should also add passed examples with a role of `columnheader` (and no `th`) to illustrate this.

Reply:
You are right @Jym77, this would be inapplicable. I have removed it completely as it matched Inapplicable example 1 (except having a role, I guess I was trying to show that just having a role on td does not suffice, but it only adds ambiguity).